skema/gromet/execution_engine/execution_engine.py

```diff
@@ -0,0 +1,198 @@
+            return self.visit_literal(node)
+        if "Primitive" in node_types:
+            return self.visit_primitive(node)
+
+    def visit_module(self, node):
+        "Visitor for top-level module"
+        node_id = node._id
+
+        expressions = self.query_runner.run_query("ordered_expressions", id=node_id)
+        for expression in expressions:
+            self.visit(expression)
+
+    def visit_expression(self, node):
+        node_id = node._id
+
+        # Only the left hand side is directly connected to the expression. So, we access the right hand side from the left hand side node
+        # (Expression) -> (Opo) -> (Primitive | Literal | Opo)
+        left_hand_side = self.query_runner.run_query("assignment_left_hand", id=node_id)
+        right_hand_side = self.query_runner.run_query(
+            "assignment_right_hand", id=left_hand_side[0]._id
+        )
+
+        # The lefthand side represents the Opo of the variable we are assigning to
+        # TODO: What if we have multiple assignment x,y = 1,2
+        # TODO: Does an expression always correspond to an assingment?
+        symbol = self.visit(left_hand_side[0])
+
+        # The right hand side can be either a LiteralValue, an Expression, or a Primitive
+        index = {"Primitive": 1, "Expression": 1, "Literal": 2}
+        right_hand_side = sorted(
+            right_hand_side, key=lambda node: index[list(node._labels)[0]]
+        )
+        value = self.visit(right_hand_side[0])
+        if not self.symbol_table.get_symbol(symbol):
+            self.symbol_table.add_symbol(symbol, value, None)
+        else:
+            self.symbol_table.update_symbol(symbol, value, None)
+
+    def visit_function(self, node):
+        """Visitor for :Opi node type"""
+        # TODO: Add support for function calls/definitions
+        pass
+
+    def visit_opo(self, node):
+        "Visitor for :Opo node type"
+        return node.name
+
+    def visit_opi(self, node):
+        """Visitor for :Opi node type"""
+        node_id = node._id
+
+        # If un-named, we need to get the name from the attached Opo
+        if node.name == "un-named":
+            return self.visit(
+                self.query_runner.run_query("assignment_left_hand", id=node_id)[0]
+            )
+
+        return node.name
+
+    def visit_literal(self, node):
+        # Convert to Tensor for execution
+
+        # TODO: Update LiteralValue to remove wrapping "" characters
+        value = node.value["value"].strip('"')
+        value_type = node.value["value_type"]
+
+        if value_type == "Integer":
+            return torch.tensor(int(value), dtype=torch.int)
+        elif value_type == "AbstractFloat":
+            return torch.tensor(float(value), dtype=torch.float)
+        elif value_type == "Boolean":
+            return torch.tensor(value == "True", dtype=torch.bool)
+        elif value_type == "List":
+            # TODO - Add support for List
+            pass
```

Comment:
should this raise and exception for the time being?

Reply:
The reason I'm not current raising an exception here is because an unsupported type isn't necessarily catastrophic. Take for example the following:  
```Python
a = 10
b = 20
c = [1,2,3] 
d = a+b
```
Here we are still able to extract parameter values for a, b, and d. With the current implementation, c will just show up as a None value. 

However, I am open to input as well.

Reply:
Thanks for the explanation.  Perhaps we should at least produce a warning?

Reply:
Yeah, I like that idea. I'll add some logging messages.